haskell/cabal.bzl

```diff
@@ -1089,7 +1089,155 @@ library
     # Sort the items to make sure that generated outputs are deterministic.
     return {name: resolved[name] for name in sorted(resolved.keys())}
 
-def _download_packages(repository_ctx, snapshot, resolved):
+def _pin_packages(repository_ctx, resolved):
+    """Pin resolved packages.
```

Comment:
Maybe a stupid question, but does this show up in the generated API docs? If not, we should probably have equivalent information somewhere `docs`. Not a blocker of course.